pkg/logs/auditor/auditor.go

```diff
@@ -267,7 +272,29 @@ func (a *RegistryAuditor) flushRegistry() error {
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(a.registryPath, mr, 0644)
+	f, err := os.CreateTemp(a.registryDirPath, a.registryTmpFile)
+	if err != nil {
+		return err
+	}
+	tmpName := f.Name()
+	defer func() {
+		if err != nil {
+			_ = f.Close()
+			_ = os.Remove(tmpName)
+		}
+	}()
+	if _, err := f.Write(mr); err != nil {
+		return err
+	}
+	if runtime.GOOS != "windows" {
+		if err := f.Chmod(0644); err != nil {
```

Comment:
Otherwise we shadow err used in the defer func and will leave the file around:
```suggestion
		if err = f.Chmod(0644); err != nil {
```